lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1725,6 +1725,28 @@ static bool sinkRescaleQuantizedNode(Function *F) {
       continue;
     }
 
+    // Combine Rescale down with Convolution node.
+    // Convolution(Rescale(X), F, B) -> Convolution(X, F, B).
+    // Convolution(X, Rescale(F), B) -> Convolution(X, F, B).
+    // Convolution(X, F, Rescale(B)) -> Convolution(X, F, B).
+    // ... and different combinations.
+    if (auto *CN = dyn_cast<ConvolutionNode>(&node)) {
+      auto *rescaleX = dyn_cast<RescaleQuantizedNode>(CN->getInput());
+      auto *rescaleF = dyn_cast<RescaleQuantizedNode>(CN->getFilter());
+      auto *rescaleB = dyn_cast<RescaleQuantizedNode>(CN->getBias());
+      auto newX = rescaleX ? rescaleX->getInput() : CN->getInput();
+      auto newF = rescaleF ? rescaleF->getInput() : CN->getFilter();
+      auto newB = rescaleB ? rescaleB->getInput() : CN->getBias();
+      if (rescaleX || rescaleF || rescaleB) {
```

Comment:
why do you need this condition?

Reply:
Without this condition convolution is going to be replaced with the same convolution every time sinkRescaleQuantizedNode is called (which is in a loop). While the transformation is correct, this will lead to node name exploding from "conv_name" to "conv_name11111111111...". Better to avoid that to make dumps more human-readable.

Reply:
ah, I overlooked :) for some reason, I thought it was the check against newX, newF etc.